### PR TITLE
docs: cleaned up code blocks and added two sentences to the intro

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR cleans up the `shell-session` blocks. Most of them were missing the prefix `$`. One of them also had multiple commands bundled into one. Additionally, I added a sentence to the intro that states the benefit of the CLI.

### CODE REVIEW

**Review by blake — COMMENTED** [review]
I assume the reason that most of the `$` signs were missing is that the code blocks do not contain command output. Per [markdown lint rule 14](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md014---dollar-signs-used-before-commands-without-showing-output), dollar signs should not be used before commands that have no output.

Our own style guide seems to conflict with this lint rule, and seems to recommend placing dollar signs in front of all shell commands.

If we're knowingly violating this lint rule and okay with that, then these changes look good to me.

**Comment by karl-cardenas-coding** [comment]
> I assume the reason that most of the `$` signs were missing is that the code blocks do not contain command output. Per [markdown lint rule 14](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md014---dollar-signs-used-before-commands-without-showing-output), dollar signs should not be used before commands that have no output.
> 
> Our own style guide seems to conflict with this lint rule, and seems to recommend placing dollar signs in front of all shell commands.
> 
> If we're knowingly violating this lint rule and okay with that, then these changes look good to me.

That's a really good call out @blake . Let me take this back to the team and see about updating the style guide. It seems like throughout the years the usage of `$` has become the standard.

**Comment by karl-cardenas-coding** [comment]
@blake I spoke to the team. And yes, we have documented in the HashiCorp markdown style guide to use `$` before all commands.  The purpose is to help differentiate the command from the output, if there is output. But the [guidance](https://github.com/hashicorp/engineering-docs/blob/main/writing/markdown.md#code-blocks) is to always use `$`.

TLDR: We can ignore this rule in the lint checker.

**Review by blake — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/605344.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 58f30a3c303c0aa8083d99b1640d2dca12af7ab6 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 58f30a3c303c0aa8083d99b1640d2dca12af7ab6 onto `release/1.11.x` succeeded!
